src/main/java/at/ac/fhcampuswien/fhmdb/HomeController.java

```diff
@@ -13,104 +13,160 @@
+    private final FilteredList<Movie> filteredMovies = new FilteredList<>(observableMovies);
+
+    /**
+     * Initializes the controller class.
+     * <p>
+     * <a href="https://docs.oracle.com/javase/8/javafx/api/javafx/fxml/Initializable.html">JavaFX Initialize</a>
+     * NOTE This interface has been superseded by automatic injection of location and resources properties into the controller.
+     * FXMLLoader will now automatically call any suitably annotated no-arg initialize() method defined by the controller.
+     * It is recommended that the injection approach be used whenever possible.
+     */
+    public void initialize() {
+        List<Movie> allMovies;
+
+        try {
+            allMovies = MovieAPIService.getMovies();
+        } catch (IOException e){
+            allMovies = new ArrayList<>();
+        }
+
         observableMovies.addAll(allMovies);
+        sortMovies();
 
-        movieListView.setItems(filteredList);
-        movieListView.setCellFactory(movieListView -> new MovieCell());
+        movieListView.setItems(filteredMovies);
+        movieListView.setCellFactory(e -> new MovieCell());
 
-        genreComboBox.setPromptText("Filter by Genre");
         genreComboBox.getItems().addAll(Genre.values());
+        genreComboBox.setValue(Genre.NO_FILTER);
 
-        releaseYearPicker.setPromptText("Filter by Release Year");
-        releaseYearPicker.getItems().addAll(getYears());
-
-        ratingComboBox.setPromptText("Filter by Rating");
-        ratingComboBox.getItems().addAll(RatingOption.values());
-
-        searchBtn.setOnAction(actionEvent -> searchForMovie(searchField.getText().trim(), genreComboBox.getValue(), filteredList, observableMovies));
+        releaseYearPicker.getItems().addAll(Year.values());
+        releaseYearPicker.setValue(Year.NO_FILTER);
 
-        sortBtn.setOnAction(actionEvent -> {
-            if(sortBtn.getText().equals("Sort (asc)")) {
-                observableMovies.sort(Comparator.naturalOrder());
-                sortBtn.setText("Sort (desc)");
-            } else {
-                observableMovies.sort(Collections.reverseOrder());
-                sortBtn.setText("Sort (asc)");
-            }
-        });
+        ratingComboBox.getItems().addAll(Rating.values());
+        ratingComboBox.setValue(Rating.NO_FILTER);
 
-        resetFilterBtn.setOnAction(actionEvent -> resetFilterCriteria(genreComboBox, filteredList, searchField));
+        searchBtn.setOnAction(actionEvent -> setFilter());
+        sortBtn.setOnAction(actionEvent -> sortMoviesAndSetButtonText());
+        resetFilterBtn.setOnAction(actionEvent -> resetFilter());
     }
 
-    public static void resetFilterCriteria(JFXComboBox<Genre> genreComboBox,
-                                           FilteredList<Movie> filteredList,
-                                           TextField searchField) {
+    private void resetFilter() {
+        observableMovies.clear();
```

Comment:
Why are you manipulating the `observableMovies` instead of the `filteredMovies`?

Reply:
Because we don't filter, we always ask the API for movies.

Reply:
Now only observableMovies are used. Filtered Movies don't exist anymore.